core/include/drivers/clk_dt.h

```diff
@@ -0,0 +1,104 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2021, Bootlin
+ */
+
+#ifndef __DRIVERS_CLK_DT_H
+#define __DRIVERS_CLK_DT_H
+
+#include <kernel/dt.h>
+#include <stdint.h>
+#include <sys/queue.h>
+
+#define MAX_CLK_PHANDLE_ARGS 3
+
+/**
+ * struct clk_dt_phandle_args - Devicetree clock args
+ * @nodeoffset: Clock consumer node offset
+ * @args_count: Count of cells for the clock
+ * @args: Clocks consumer specifiers
+ */
+struct clk_dt_phandle_args {
+	int nodeoffset;
+	int args_count;
+	uint32_t *args;
+};
+
+/**
+ * struct clk_driver - clk driver setup struct
+ * setup:
+ */
+struct clk_driver {
+	TEE_Result (*setup)(const void *fdt, int nodeoffset);
+};
+
+/**
+ * CLK_DT_DECLARE - Declare a clock driver
+ * @__name: Clock driver name
+ * @__compat: Compatible string
+ * @__init: Clock setup function
```

Comment:
`__setup` would be better argument name IMO.

Reply:
Yes, fixed.